clang/lib/CConv/ArrayBoundsInferenceConsumer.cpp

```diff
@@ -617,6 +619,183 @@ void AddMainFuncHeuristic(ASTContext *C, ProgramInfo &I, FunctionDecl *FD) {
 
 }
 
+// Given a variable I, this visitor collects all the variables that are used as
+// RHS operand of < and I >=  expression.
+// i.e., for all I < X expressions, it collects X.
+class ComparisionVisitor : public StmtVisitor<ComparisionVisitor> {
```

Comment:
Why is `StmtVisitor` used instead of a `RecursiveASTVisitor`? I don't think I fully understand the difference, but I feel like the ast visitor might let you remove the boiler plate `VisitStmt` and such.

Reply:
You are right. Made it a `RecursiveASTVisitor`.